elastic/kibana
# [Security Solution] Add rule upgrade review cache invalidation

**Resolves: https://github.com/elastic/kibana/issues/202779**

## Summary

Implemented proper cache invalidation after rule CRUD operations to prevent displaying stale data in rule upgrade flyouts. See the linked issue for more details.

### How to Test

1. Ensure rule customization is enabled
2. Navigate to the Rule Updates page 
3. Locate any rule with updates
4. Make changes to the rule:
   - Edit the rule via the Rule Editing page
   - Use bulk actions
   - Delete the rule
5. Return to the Rule Updates page
   - All introduced changes should be reflected:
     - Modifications should be visible
     - If the rule was deleted, it should no longer appear on the page
